### ISSUE TITLE
Optimize memory usage and performance in SD3 inference pipeline

### ISSUE THREAD

**Issue by NewAITees** [NewAITees/GenerativeAIArtWeb#5]
## Problem
The current implementation in `src/generator/sd3_inf.py` doesn't optimize memory usage efficiently, which can lead to high VRAM consumption and potential out-of-memory errors during image generation, especially with larger resolution images or batch processing.

## Proposed Solution
1. Implement memory optimization strategies for model loading and inference
2. Add efficient tensor management during generation process
3. Optimize CUDA memory handling with proper garbage collection
4. Implement model weight offloading for CPU-GPU transfer when needed

## Implementation Approach
1. Add targeted CUDA memory clearing:
   ```python
   import gc
   import torch
   
   def clear_cuda_memory():
       torch.cuda.empty_cache()
       gc.collect()
   ```

2. Implement gradient-free inference with context manager:
   ```python
   @contextmanager
   def inference_mode():
       with torch.inference_mode():
           with torch.cuda.amp.autocast(enabled=True):
               yield
   ```

3. Add efficient tensor moving between CPU and GPU:
   ```python
   def to_device(tensor, device):
       if tensor.device == device:
           return tensor
       return tensor.to(device)
   ```

4. Implement progressive model loading for large models

5. Add adaptive batch size based on available VRAM

## Benefits
- Higher resolution image generation with the same hardware
- Lower memory footprint during inference
- Fewer out-of-memory errors
- Ability to handle larger batch sizes
- Better user experience on machines with limited VRAM
